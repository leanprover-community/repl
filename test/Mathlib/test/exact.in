{"cmd": "import Mathlib"}

{"cmd": "theorem test : 0 < 1 := by sorry", "env": 0, "sorries": "individual"}

{"tactic": "exact?", "proofState": 0}

{"cmd": "theorem test : 3 = 7 := by sorry", "env": 0, "sorries": "individual"}

{"tactic": "exact?", "proofState": 2}

