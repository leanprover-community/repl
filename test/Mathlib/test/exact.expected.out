{"env": 0}

{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 27},
   "goal": "⊢ 0 < 1",
   "endPos": {"line": 1, "column": 32}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 12},
   "data": "declaration uses 'sorry'"}],
 "env": 1}

{"proofStatus": "Completed",
 "proofState": 1,
 "messages":
 [{"severity": "info",
   "pos": {"line": 0, "column": 0},
   "endPos": {"line": 0, "column": 0},
   "data": "Try this:\n  exact Nat.one_pos"}],
 "goals": []}

{"sorries":
 [{"proofState": 2,
   "pos": {"line": 1, "column": 27},
   "goal": "⊢ 3 = 7",
   "endPos": {"line": 1, "column": 32}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 12},
   "data": "declaration uses 'sorry'"}],
 "env": 2}

{"message":
 "Lean error:\n`exact?` could not close the goal. Try `apply?` to see partial suggestions."}

